zk-token-sdk/src/instruction/transfer/without_fee.rs

```diff
@@ -96,7 +97,8 @@ impl TransferData {
         (destination_pubkey, auditor_pubkey): (&ElGamalPubkey, &ElGamalPubkey),
     ) -> Result<Self, ProofGenerationError> {
         // split and encrypt transfer amount
-        let (amount_lo, amount_hi) = split_u64(transfer_amount, TRANSFER_AMOUNT_LO_BITS);
+        let (amount_lo, amount_hi) = try_split_u64(transfer_amount, TRANSFER_AMOUNT_LO_BITS)
+            .map_err(|_| ProofGenerationError::IllegalAmountBitLength)?;
```

Comment:
So just to make sure, `TransferData::new` is only called in token-2022, and not in the zk proof program, which is why it's safe to make this change directly. Is that correct?

Reply:
Oh sorry, it is not in the same file, but in `with_fee.rs`: https://github.com/solana-labs/solana/pull/34130/files#diff-519c9b6865518cc65f5f34c248759612113b0b2d5ad90e3bba9729c31b1b1309R693-R704.